challenge_7/java/jdfurlan/src/FindTheMissingNumber.java

```diff
@@ -0,0 +1,21 @@
+
+public class FindTheMissingNumber {
+
+	public static void main(String[] args) {
+		int[] arr = { 6, 3, 4, 8, 1, 2, 0, 9, 10, 7 }; // missing is 5
+		System.out.println(findMissingNum(arr));
+	}
+
+	private static int findMissingNum(int[] arr) {
+		int actualSum = 0;
+		int thisSum = 0;
+		int i;
+		for (i = 0; i < arr.length; i++) {
+			actualSum += i;
+			thisSum += arr[i];
+		}
```

Comment:
Hmm. This looks an off by one error. If your array is `[1, 2, 3]` what will `actualSum` be? 5. What will `thisSum` be? 6. What's `actualSum - thisSum`? -1.  Have zero be the missing number and run this application again.